tools/importer-rest-api-specs/internal/components/apidefinitions/parser/parsingcontext/build.go

```diff
@@ -0,0 +1,330 @@
+// the parsed specs for consumption in the Resource Manager parser.
+//
+// We are intentionally parsing each swagger file twice and flattening it with different options.
+// On the first pass, the file is flattened with the `Minimal: true` option, which  leaves `allOf` refs in place
+// and does not inline properties from inherited models. This is the `SwaggerSpecWithReferences` version, and
+// allows us to inspect the inheritance tree and properly derive discriminated models.
+//
+// Note that we also export the `spec.Swagger` struct containing the raw spec. This is currently used in a few
+// cases where we want to manually parse the definitions, notably for traversing model inheritance for
+// discriminated models.
+//
+// On the second pass, we flatten with `Minimal: false`, inlines all refs including `allOf`. This is the
+// `SwaggerSpecExpanded` version, and is useful because models that inherit from parents will/ have all their
+// fields collapsed into that model, making it easier for us to iterate them.
+//
+// Additionally, on each pass of parsing a swagger file, we iterate the definitions found within it, and manually
+// resolve all remote references. This works around buggy behavior in the go-openapi modules where relative paths
+// are mangled and thus fail to be ingested. To prevent accidental clobbering of models/constants in external swagger
+// files that have the same name, we specifically only compile the models/constants that are being remotely referenced,
+// as well as any references they may have to ensure no reference remains unresolved. To trick the `analysis.Flatten`
+// function, we strip out the paths in remove references after resolving them, to make it appear like everything was
+// loaded from the original swagger file.
+func BuildFromFile(filePath string) (*Context, error) {
+
+	// 1. Parse the file, then resolve all remote refs, and flatten without inlining `allOf` references
+
+	// load and parse the spec
+	swaggerDocWithReferences, err := loads.Spec(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("loading swagger file %q: %+v", filePath, err)
+	}
+
+	// walk the refs in the loaded spec and resolve any external references
+	specWithRemotelyReferencedSchemas, err := resolveExternalSwaggerReferences(swaggerDocWithReferences, filePath)
+	if err != nil {
+		return nil, fmt.Errorf("could not resolve remote swagger references for %q: %+v", filePath, err)
+	}
+
+	// mix-in the remotely referenced schemas from any external swagger documents,
+	analysis.Mixin(swaggerDocWithReferences.Spec(), specWithRemotelyReferencedSchemas)
+
+	// flatten the spec, whilst retaining refs instead of inlining them
+	basePath := swaggerDocWithReferences.SpecFilePath()
+	flattenedWithReferencesOpts := &analysis.FlattenOpts{
+		Minimal:      true,
+		Verbose:      true,
+		Expand:       false,
+		RemoveUnused: false,
+		//ContinueOnError: true,
```

Comment:
Should this still be commented out?

Reply:
I left this in because it might be useful to uncomment to have the parser fail on warnings. will add an explanatory comment